be/src/storage/task/engine_clone_task.cpp

```diff
@@ -445,7 +445,32 @@ Status EngineCloneTask::_download_files(DataDir* data_dir, const std::string& re
         return Status::OK();
     };
     RETURN_IF_ERROR(HttpClient::execute_with_retry(DOWNLOAD_FILE_MAX_RETRY, 1, list_files_cb));
-    std::vector<string> file_name_list = strings::Split(file_list_str, "\n", strings::SkipWhitespace());
+
+    // Parse file name and size
+    const std::string FILE_DELIMETER_IN_DIR_RESPONSE = "\n";
+    const std::string FILE_NAME_SIZE_DELIMETER = "|";
+    std::vector<string> file_name_list;
+    std::vector<int64_t> file_size_list;
+
+    bool use_file_name_and_size_format = file_list_str.find(FILE_NAME_SIZE_DELIMETER) != string::npos;
+    if (use_file_name_and_size_format) {
+        for (auto file_str : strings::Split(file_list_str, FILE_DELIMETER_IN_DIR_RESPONSE, strings::SkipWhitespace())) {
+            std::vector<string> list = strings::Split(file_str, FILE_NAME_SIZE_DELIMETER);
+            if (list.size() != 2) {
+                continue;
+            }
+
+            int64_t file_size = atoll(list[1].c_str());
```

Comment:
`atoll` does not detect errors, better use one of the functions in `be/src/util/string_parser.hpp`